mitodl/mitxonline
# Enrollment dialog WCAG fixes

# What are the relevant tickets?

Closes mitodl/hq#2616

# Description (What does it do?)

Makes a bunch of changes to the enrollment upsell modal and the more dates selector:

**More Dates**
- No longer clickable - you can view the dates but it's just informational now.
- Enrolled course runs should now appear green with the "Enrolled" text if you're enrolled in the course. (This was there, but wasn't working - it now considers your actual enrollments rather than rely on the courserun API.)

**Upsell Modal**
- If there's an upgrade available, clicking Enroll Now won't immediately enroll you in the course (at all)
- If there are multiple course runs, you should be able to select what course run you want to enroll in
- Selecting a different course run should update the modal text appropriately (upgrade deadlines, prices, etc.) 
- Selecting the "Enroll and Pay" button should drop you into the cart
- Selecting "Enroll for Free" should just enroll you in the course and should bring you to the dashboard as per usual
- If financial assistance is available, you should see "Need financial assistance?"
- Clicking the "Need financial assistance?" link should bring you to the appropriate financial assistance form
- Closing the upsell modal should leave you on the page - you should _not_ be enrolled in the course and you should not go to the dashboard, as you did before
- If the course has multiple course runs and you're enrolled in one of them, the one you're enrolled in should be selected

# Screenshots (if appropriate):

Date selector version:
![image](https://github.com/mitodl/mitxonline/assets/945611/5cd0cdf9-0385-4439-900b-91d5e44e4c06)
Date selector version with a payment deadline:
![image](https://github.com/mitodl/mitxonline/assets/945611/eee18949-ac50-40fd-94cb-6f6819c7ed16)
No date selector:
![image](https://github.com/mitodl/mitxonline/assets/945611/e61a1e9e-3b5c-497c-851e-1d8a51905580)

# How can this be tested?

Automated tests should pass.

Selecting More Dates in the Course Info Box on a course with multiple course runs should display the dates. Selecting individual dates should have no effect, and runs that you're enrolled in should have "<date> - Enrolled" text and be presented in green.

Selecting Enroll Now on a course with a single active course run and a product should pop the upsell modal (see screenshot). You should not see the date selector. An enrollment should not be created when you see the modal. Clicking Enroll and Pay should allow you to pay for the course. Clicking Enroll for Free should enroll you in the course and deliver you to the dashboard. Clicking Need financial assistance? (if available) should bring you to the financial assistance form. Clicking the close button or outside the modal should close the modal, and you should remain on the course detail page; again, no enrollment should be created.

Selecting Enroll Now on a course with multiple active course runs (with products) should pop the upsell modal. You should see the date selector, and choosing different dates should update the modal as appropriate. (For testing, it would be a good idea to have at least one with a payment deadline.) The modal should otherwise work like the version for a single active course run.

# Additional Context

Some styling changes were made to select controls and to modal headers that are global - the design indicates that there were changes here so I applied them globally. 

This needs some more tests to cover some scenarios but will do that in a separate PR so we can get this out the door.
